### ISSUE TITLE
M

### ISSUE THREAD

**Issue by phorn-NI** [approvals/ApprovalTests.LabVIEW#4]
Caraya supports a margin of error when comparing floating point numbers (as 10.0 != 10.0 exactly)

I found a similar issue when generating png images and trying to compare them. The two attached images were generated by exactly the same configuration script and look identical, but not when you put them through a compare tool - however for my purposes they're close enough (its the ends of the error bars that are 1 or 2 pixels different).

Its not urgent as the work around is to compare the configuration script, which is generated from a list of parameters - and that's what I'm really trying test
![Test error-bar override test error bar base json formatter approved](https://github.com/user-attachments/assets/4bf3d7a1-0be8-40ca-bb21-a417b8178ba2)
![Test error-bar override test error bar base json formatter received](https://github.com/user-attachments/assets/c696ef26-1916-4836-b17a-b1c2d569909d)

**Comment by stagg54** [comment]
Hmm... I imagine this is fixable, but it will require some research. There is probably a png compare tool out there that might work for something like this. Right now IIRC (without looking at the code) I'm pretty sure it's just checking the file hashes to see if they match. That would have to tolerance builtin. That comparison is overrideable though - for VI and CTL files, it uses the LVCompare function.

**Comment by phorn-NI** [comment]
Interesting research project then 😄 
As I said, it's not a big problem for my immediate use case - testing the generated json is robust and gives me the same info.

**Comment by stagg54** [comment]
Yeah I did a very brief look. ImageMagick seems to have a compare tool. Not sure how well that would work.

**Comment by stagg54** [comment]
that should work for images. For numeric stuff, you could write a printer and run your object through that first to turn it into a string and then verify the string.
The printer could just round everything o to some arbitrary precision. Round all floats to 3 decimal places or something.
